### PR TITLE
447 create sprint template test

### PR DESCRIPTION
**Pull request by jeremydthomas**
Pull-Request for `paretOS`

## Description
added e2e test with puppeteer, from login to logout

## Relates to
- #452  
## Reviewers
- @person1 (merge duty)
- @person2

### CODE REVIEW

File: tests/integration/CreateSprintTemplate.test.ts
Comment (jayeclark) [inline tests/integration/CreateSprintTemplate.test.ts:67]: This can come out, right? ob is all we need - when the first item gets dragged to a new column, another one will take its place. 

Maybe we could have more descriptive variables for these items? like
optionsColumnItem1BoundingBox, morningColumnBoundingBox, workDayColumnBoundingBox, eveningColumnBoundingBox.  It's a little verbose but would help make it really clear what's happening.

File: tests/integration/CreateSprintTemplate.test.ts
Comment (jayeclark) [inline tests/integration/CreateSprintTemplate.test.ts:62]: Don't need origin2 or origin3, right? (See later comment.)

File: tests/integration/CreateSprintTemplate.test.ts
Comment (jayeclark) [inline tests/integration/CreateSprintTemplate.test.ts:60]: Just FYI - if you want, you can get rid of a lot of the ?'s and !'s later on if you cast the elements and their bounding boxes to the specific types, ie

```typescript
const origin1 = await page.waitForSelector(`[id="Talk to Customers"]`) as ElementHandle<Element>;
```

and

```typescript
const ob = await origin1?.boundingBox() as BoundingBox;
```

File: tests/integration/CreateSprintTemplate.test.ts
Comment (jayeclark) [inline tests/integration/CreateSprintTemplate.test.ts:75]: See earlier comment about casting to types, could help clean up this section & all the comments telling DeepSource to ignore the issues :)

File: tests/integration/CreateSprintTemplate.test.ts
Comment (jayeclark) [inline tests/integration/CreateSprintTemplate.test.ts:76]: If you're planning to keep these in here, would be helpful to add some description so it makes sense to new people running the tests. IE 

```typescript
`Dragging from x=${ob!?.x + ob!.width / 2}, y=${ob!.y + ob!.height / 2} (coordinates of first item in options column)`
```

File: tests/integration/CreateSprintTemplate.test.ts
Comment (jayeclark) [inline tests/integration/CreateSprintTemplate.test.ts:31]: Should probably log out before closing the browser?

**Review by jayeclark — COMMENTED** [review]
Looks good! I left a bunch of stylistic/readability comments and also some TypeScript tips. They're mostly suggestions on how to make this more navigable for people new to the repository (so if they make changes that break a test they can more easily figure out why they might have broken the test.)

_General comment:_ these new integration tests are failing about 1/4 of the time on my local machine. Seems to be related specifically to lines 74-77 and especially ob.width. I suspect it might be related to the fact that logging in is its own test in the series rather than a prerequisite in beforeAll. Will have to see if the same behavior is observed in CI/CD (only time will tell on that.)

Can you add a commit (or amend the last one) and add 'ignore' comments for the remaining DeepSource issues? Once all the checks are green I'm happy to merge :) or you can let me know if you want to address any of the comments.

File: tests/integration/CreateSprintTemplate.test.ts
Comment (jeremydthomas) [reply tests/integration/CreateSprintTemplate.test.ts:31]: Ya logout is the final thing before it takes a screenshot if it fails, I can move it up there if you want to.

**Comment by jeremydthomas** [comment]
I made the changes, I probably missed one or more. Let me know how else I can approve it. Was sneaking this in between work tasks 😆

File: tests/integration/CreateSprintTemplate.test.ts
Comment (jayeclark) [inline tests/integration/CreateSprintTemplate.test.ts:81]: Thanks for updating these, this makes it a lot more readable. It's easier to follow along with what the test is doing without having to remember what ob or mb stands for. 🙂

File: tests/integration/CreateSprintTemplate.test.ts
Comment (jayeclark) [inline tests/integration/CreateSprintTemplate.test.ts:66]: Looks a lot cleaner with the type casting in here (and the DeepSource ignore comments & !?'s removed!

File: tests/integration/CreateSprintTemplate.test.ts
Comment (jayeclark) [inline tests/integration/CreateSprintTemplate.test.ts:13]: Yeah, this is much clearer - it's easier to see what settings one can fiddle with to adjust the testing conditions and the explanatory comments are all at the start of the file rather than crowding the actual code execution.

**Review by jayeclark — CHANGES_REQUESTED** [review]
Looks great @jeremydthomas ! I think the only thing remaining before I can approve to merge is the DeepSource issues. Looks like there are just a few of them, mainly related to LeftNav. 

LMK if you want me to just take care of them (I think you can check the "allow maintainers to edit" box and that lets me actually update your code directly.) Adding ignore comments to issues in JSX is a little tricky.

**Comment by jeremydthomas** [comment]
I must be blind here, I dont see the part where i can let you do it.

**Comment by jayeclark** [comment]
Should be a checkbox for you in the right hand column...
Screen Shot 2023-02-26 at 3 50 11 PM

**Comment by jayeclark** [comment]
Looks like I already have access to edit 🙌

**Review by jayeclark — APPROVED** [review]

**Comment by jayeclark** [comment]
🙌 🙌 🙌 
Screen Shot 2023-02-26 at 4 36 01 PM

Approved & merged, @jeremydthomas !

**Pull request merged**
